dspace-api/src/test/java/org/dspace/content/RelationshipServiceImplPlaceTest.java

```diff
@@ -0,0 +1,374 @@
+
+        Relationship relationship = relationshipService.create(context, item, authorItem, isAuthorOfPublication, 2, -1);
+
+        context.restoreAuthSystemState();
+
+        List<MetadataValue> list = itemService
+            .getMetadata(item, dcSchema, contributorElement, authorQualifier, Item.ANY);
+        assertThat(list.size(), equalTo(3));
+
+        assertMetadataValue(authorQualifier, contributorElement, dcSchema, "test, one", null, 0, list.get(0));
+        assertMetadataValue(authorQualifier, contributorElement, dcSchema, "test, two", null, 1, list.get(1));
+        assertMetadataValue(authorQualifier, contributorElement, dcSchema, "familyName, firstName",
+                            "virtual::" + relationship.getID(), 2, list.get(2));
+        assertThat(relationship.getLeftPlace(), equalTo(2));
+
+        context.turnOffAuthorisationSystem();
+
+        itemService.addMetadata(context, item, dcSchema, contributorElement, authorQualifier, null, "test, three");
+        itemService.addMetadata(context, item, dcSchema, contributorElement, authorQualifier, null, "test, four");
+
+        WorkspaceItem authorIs = workspaceItemService.create(context, col, false);
+        Item secondAuthorItem = installItemService.installItem(context, authorIs);
+        itemService.addMetadata(context, secondAuthorItem, "relationship", "type", null, null, "Person");
+        itemService.addMetadata(context, secondAuthorItem, "person", "familyName", null, null, "familyNameTwo");
+        itemService.addMetadata(context, secondAuthorItem, "person", "givenName", null, null, "firstNameTwo");
+        Relationship relationshipTwo = relationshipService
+            .create(context, item, secondAuthorItem, isAuthorOfPublication, 5, -1);
+
+        context.restoreAuthSystemState();
+
+        list = itemService.getMetadata(item, dcSchema, contributorElement, authorQualifier, Item.ANY);
+
+        assertMetadataValue(authorQualifier, contributorElement, dcSchema, "test, one", null, 0, list.get(0));
+        assertMetadataValue(authorQualifier, contributorElement, dcSchema, "test, two", null, 1, list.get(1));
+        assertMetadataValue(authorQualifier, contributorElement, dcSchema, "familyName, firstName",
+                            "virtual::" + relationship.getID(), 2, list.get(2));
+        assertThat(relationship.getLeftPlace(), equalTo(2));
+
+        assertMetadataValue(authorQualifier, contributorElement, dcSchema, "test, three", null, 3, list.get(3));
+        assertMetadataValue(authorQualifier, contributorElement, dcSchema, "test, four", null, 4, list.get(4));
+        assertMetadataValue(authorQualifier, contributorElement, dcSchema, "familyNameTwo, firstNameTwo",
+                            "virtual::" + relationshipTwo.getID(), 5, list.get(5));
+        assertThat(relationshipTwo.getLeftPlace(), equalTo(5));
+
+    }
+
+
+    @Test
+    public void AddAndRemoveMetadataAndRelationshipsTest() throws Exception {
```

Comment:
Same as above.  Too large and no comments

Reply:
Inline comments have been included